src/core/meshcop/border_agent.cpp

```diff
@@ -670,6 +670,7 @@ void BorderAgent::HandleConnected(SecureTransport::ConnectEvent aEvent)
         if (mUsingEphemeralKey)
         {
             mCounters.mEpskcSecureSessionSuccesses++;
+            mEphemeralKeyTask.Post();
```

Comment:
I wonder if it's possible to pass some args in the ephemeral key callback, for example the epskc string, the timeout value, wether it has secure session connected, etc. Thinking of the future where multiple epskc sessions are supported, if one of them has an active secure session, we may want to stop advertising `_meshcop-e` for that epskc only?

Reply:
I don't think we would support multiple e-PSK sessions at the same time. We might consider allowing multiple connections using the same PSKc in the future, though I'm not convinced it's the best approach. If we choose this direction, many `otBorderAgent` APIs would need redesigning, as the current `otBorderAgentGetState()` inherently assumes a single connection/session.

> I wonder if it's possible to pass some args in the ephemeral key callback

It is technically possible, but it would duplicate information already available through the existing APIs. My preference is to provide one way to achieve any functionality, to help keep the code simpler and easier to test/check. 

In next-layer code (e.g., `ot-br-posix`), we can  determine all necessary information by checking different states/APIs (e.g., combination of `otBorderAgentIsEphemeralKeyActive()` and `otBorderAgentGetState()`). This can simplify the code which decides about publishing/unpublishing the "_meshcop-e._udp" service vs "_meshcop._udp" service, as this code only needs to check those APIs. This allows us to reuse the same code/method regardless of if it is called from the `otBorderAgentEphemeralKeyCallback` or from another code path.

Reply:
It's mentioned [here](https://github.com/openthread/openthread/pull/10672#discussion_r1746503079) by @jwhui that multiple secure session each with its own ephemeral key (and maybe also PSKc) may be considered for the future.

It's not clear to me how that would be supported, to me if we want to start to consider that, the ephemeral key callback needs to at least return the key string(or a key id) to identify which key has a state update. Thoughts?

For the current model where only one session is supported, this change LGTM.